### ISSUE TITLE
ConcurrentModificationException from 1.3.9

### ISSUE THREAD

**Issue by wirefreethought** [bohnman/squiggly-java#36]
I'm getting the following stack from 1.3.9 when attempting to invoke ObjectMapper.writeValueAsString (works fine in 1.3.8 and earlier):
```
Caused by: java.util.ConcurrentModificationException: null
	at java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:719) ~[na:1.8.0_162]
	at java.util.LinkedHashMap$LinkedValueIterator.next(LinkedHashMap.java:747) ~[na:1.8.0_162]
	at com.github.bohnman.squiggly.parser.SquigglyParser.analyze(SquigglyParser.java:207) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.parser.SquigglyParser.access$100(SquigglyParser.java:37) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.parser.SquigglyParser$Visitor.visitParse(SquigglyParser.java:88) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.parser.SquigglyParser$Visitor.visitParse(SquigglyParser.java:83) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.parser.antlr4.SquigglyExpressionParser$ParseContext.accept(SquigglyExpressionParser.java:113) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at org.antlr.v4.runtime.tree.AbstractParseTreeVisitor.visit(AbstractParseTreeVisitor.java:18) ~[antlr4-runtime-4.6.jar:4.6]
	at com.github.bohnman.squiggly.parser.SquigglyParser.parse(SquigglyParser.java:73) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.context.LazySquigglyContext.getNodes(LazySquigglyContext.java:34) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.filter.SquigglyPropertyFilter.pathMatches(SquigglyPropertyFilter.java:181) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.filter.SquigglyPropertyFilter.include(SquigglyPropertyFilter.java:169) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.github.bohnman.squiggly.filter.SquigglyPropertyFilter.serializeAsField(SquigglyPropertyFilter.java:338) ~[squiggly-filter-jackson-1.3.9.jar:na]
	at com.fasterxml.jackson.databind.ser.std.BeanSerializerBase.serializeFieldsFiltered(BeanSerializerBase.java:742) ~[jackson-databind-2.8.10.jar:2.8.10]
```

**Comment by bohnman** [comment]
@wirefreethought Can you provide the filter expression you were using?

**Comment by wirefreethought** [comment]
@bohnman ``-data.country*,-data.region*,links*,metadata*``

**Comment by bohnman** [comment]
@wirefreethought I released version 1.3.10, which should fix this issue.  Please close if you are satisfied with the fix.

**Comment by wirefreethought** [comment]
@bohnman Fixed. Thanks for the quick turnaround.
